java/system-test/system-test-configuration/src/main/java/sleeper/systemtest/configuration/SystemTestPropertyImpl.java

```diff
@@ -96,6 +97,17 @@ public String toString() {
         return propertyName;
     }
 
+    @Override
+    public String computeValue(String value, InstanceProperties instanceProperties) {
+        if (value != null) {
+            return value;
+        } else if (getDefaultValue() != null) {
+            return getDefaultValue();
+        } else {
+            return null;
+        }
```

Comment:
The second if is unnecessary, as we can just return the result of `getDefaultValue()`.

This is also missing unit tests. Some options would be:
1. Share the logic with UserDefinedInstancePropertyImpl somehow, e.g. by moving the code to InstanceProperties calling getters
2. Test this in SystemTestPropertiesTest

Reply:
It also looks like changes in this PR have an effect on SystemTestStandaloneProperties, because that doesn't override the get method on SleeperProperties. Maybe we should make SleeperProperties.get abstract, and override it to call computeValue from SystemTestStandaloneProperties.get as well? It might be worth unit testing that.

Reply:
Addressed first comment